src/xdp/program.c

```diff
@@ -691,6 +691,9 @@ XdpParseFrame(
             goto BufferTooSmall;
         }
         Cache->Ip4Hdr = (IPV4_HEADER *)&Va[Offset];
+        if (Cache->Ip4Hdr->HeaderLength != sizeof(*Cache->Ip4Hdr)) {
```

Comment:
The header length field needs to be multiplied by 4, right?

Reply:
Correct, sorry for that!